### ISSUE TITLE
Build Nut for OSX: build core.go:9:5: cannot find package "github.com/..."

### ISSUE THREAD

**Issue by mattscilipoti** [matthieudelaro/nut#18]
While attempting to follow "Compiling from source" instructions, step #3 (Build Nut for OSX) cannot find the dependent packages.

Example:
```
core.go:9:5: cannot find package "github.com/Sirupsen/logrus" in any of:
	/go/src/github.com/matthieudelaro/nut/vendor/github.com/Sirupsen/logrus (vendor tree)
	/usr/local/go/src/github.com/Sirupsen/logrus (from $GOROOT)
	/go/src/github.com/Sirupsen/logrus (from $GOPATH)
init.go:6:5: cannot find package "github.com/codegangsta/cli" in any of:
	/go/src/github.com/matthieudelaro/nut/vendor/github.com/codegangsta/cli (vendor tree)
	/usr/local/go/src/github.com/codegangsta/cli (from $GOROOT)
	/go/src/github.com/codegangsta/cli (from $GOPATH)
core.go:6:5: cannot find package "github.com/fsouza/go-dockerclient" in any of: 
...
```

**Comment by mattscilipoti** [comment]
Cause: the dependencies were not in the 'vendor' dir, because I did not use `--recursive` flag for `git clone` (as indicated in the instructions).  I had cloned the repo prior to starting these instructions and did not notice the additional flag.

```
git clone https://github.com/matthieudelaro/nut.git --recursive
```

**Comment by mattscilipoti** [comment]
I am new to Go.  Is this a common practice for Go projects?  Would you expect anyone who works with Go to be aware of this or should there be a note in the readme?

**Comment by matthieudelaro** [comment]
Hi,

I think it is a quite common practice. Another one is to use a dependency manager such as godep. We could add this one to the repo, but then we should ship a docker image with godep installed.

The README explicitly says to use the flag --recursive. so I'm not sure I understand what you mean. Do you mean,  adding a comment explaining the benefit of the flag --recursive ?

**Comment by mattscilipoti** [comment]
Thanks for the info.  I did a lot of searching for the answer to the missing dependencies.  None of it led me to the `--recursive` flag.  I was just lucky that I started from scratch, following your instructions, and noticed extra directories during the clone.  As I said, previously I had cloned it prior to reading the instructions. In fact, I was reading them from a local file.  

Since this is a common practice with Go, I am leaning toward letting others struggle with this too.  It forced me to learn more about Go and it is a lesson I won't soon forget.
